llvm/lib/CodeGen/StackMaps.cpp

```diff
@@ -591,9 +591,10 @@ void StackMaps::recordStackMapOpers(
   std::set<int64_t> TrackedRegisters;
 
   // Patchpoints and stackmaps have different instruction formats.
-  // Patchpoints embed function call arguments directly into the
-  // instruction, while stackmaps are simpler and only contain the
-  // stackmap data - therefore we need to treat them differently.
+  // Patchpoints embed function and its arguments directly into the
```

Comment:
Ah that's better. We can simplify this a bit, e.g. "When collecting tracked registers from patchpoints we need to be careful that we don't just blindly copy over operands. Unlike stackmaps, patchpoints contain an embedded call and arguments which we must not track."

Reply:
updated in :point_right: 4fd45ba6f1ad1a842ffce468870fc3065a3c3f5f